lib/features/home/presentation/widgets/full_screen_reel_view.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:flutter/material.dart';
+import 'package:video_player/video_player.dart';
+
+class FullScreenReelView extends StatefulWidget {
+  const FullScreenReelView({
+    required this.videoPath,
+    super.key,
+  });
+  final String videoPath;
+
+  @override
+  // ignore: library_private_types_in_public_api
+  _FullScreenReelViewState createState() => _FullScreenReelViewState();
+}
+
+class _FullScreenReelViewState extends State<FullScreenReelView> {
+  late VideoPlayerController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = VideoPlayerController.asset(widget.videoPath)
+      ..initialize().then((_) {
+        setState(() {
+          _controller
+            ..play()
+            ..setLooping(true);
+        });
+      });
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: GestureDetector(
```

Comment:
```suggestion
      body: GestureDetector(
```
First of all this widget should be a vertically scrolling pageview so that we can scroll vertically

Reply:
Upon inspecting the code  and trying out some things i found that,

```late VideoPlayerController _controller;
  Timer? _timer;

  @override
  void initState() {
    super.initState();
    _controller = VideoPlayerController.asset(widget.videoPath)
      ..initialize().then((_) {
        _controller
          ..setLooping(false) // Disable default looping
          ..play();
        setState(() {}); // Update the UI
        _startLoop();
      });
  }

  void _startLoop() {
    _timer = Timer.periodic(const Duration(seconds: 3), (Timer timer) {
      if (_controller.value.isInitialized) {
        _controller
          ..seekTo(Duration.zero) // Reset video to start
          ..play(); // Play again
      }
    });
  }

  @override
  void dispose() {
    _timer?.cancel();
    _controller.dispose();
    super.dispose();
  }```

If we comment this code out from reel_video_card.dart and modify some intialisation logic inside the full screen_reel view with this


``` class FullScreenReelView extends StatefulWidget {
  const FullScreenReelView({
    required this.videoPath,
    super.key,
  });

  final String videoPath;

  @override
  // ignore: library_private_types_in_public_api
  _FullScreenReelViewState createState() => _FullScreenReelViewState();
}

class _FullScreenReelViewState extends State<FullScreenReelView> {
  late VideoPlayerController _controller;
  late Future<void> _initializeFuture;
  late final GlobalKey _key = GlobalKey();

  @override
  void initState() {
    super.initState();
    _controller = VideoPlayerController.asset(widget.videoPath)
      ..setLooping(true)
      ..play();
    _initializeFuture = _controller.initialize();
  }

  @override
  void dispose() {
    _controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(),
      // 1 for sample
      body: PageView.builder(
        itemCount: 1,
        key: _key,
        scrollDirection: Axis.vertical,
        itemBuilder: (_, int index) => _buildVideoView(context),
      ),
    );
  }

  Widget _buildVideoView(BuildContext context) {
    return GestureDetector(
      onTap: () => _controller.value.isPlaying
          ? _controller.pause()
          : _controller.play(),
      child: Center(
        child: FutureBuilder<void>(
          future: _initializeFuture,
          builder: (BuildContext context, AsyncSnapshot<void> snapshot) {
            if (snapshot.connectionState == ConnectionState.done) {
              return AspectRatio(
                aspectRatio: _controller.value.aspectRatio,
                child: VideoPlayer(_controller),
              );
            } else {
              return const SizedBox.square(
                dimension: 30,
                child: CircularProgressIndicator(),
              );
            }
          },
        ),
      ),
    );
  }
}``` 


the inside video seems to loop perfectly fine. this means there is something going on with the timer or the controller that is using the cached instance out on the home screen

Reply:
The function that is running on the home screen needs to be thought upon to play it for some first 30% of the total length of the video @ritesh-chaudhari-aepl